backend/sirius-components-compatibility/src/main/java/org/eclipse/sirius/components/compatibility/services/diagrams/ToolProvider.java

```diff
@@ -362,7 +361,7 @@ private CreateEdgeTool convertEdgeCreationDescription(Map<UUID, NodeDescription>
         // @formatter:on
     }
 
-    private DeleteTool convertDeleteElementDescription(Map<UUID, NodeDescription> id2NodeDescriptions, AQLInterpreter interpreter, DeleteElementDescription deleteElementDescription) {
+    private CreateNodeTool convertDeleteElementDescription(Map<UUID, NodeDescription> id2NodeDescriptions, AQLInterpreter interpreter, DeleteElementDescription deleteElementDescription) {
```

Comment:
We should probably rename `CreateNodeTool` into simply `NodeTool`? (and similarly for edges)

Reply:
[#1022](https://github.com/eclipse-sirius/sirius-components/issues/1022) will handle that.

Reply:
Yep and I think that it should not even be a `NodeTool`, it's just a tool in the palette activated by a single click as opposed to tools that are activated in two steps by opening the palette on the source and then selecting the target. Technically what we are calling today a `CreateNodeTool` can be applied on an edge.

I think I would be in favor of something radical like `SingleSelectionTool` (for the current `CreateNodeTool`) and `DoubleSelectionTool` (for the current `CreateEdgeTool`) which could then trigger a `SinglePositionEvent` and a `DoublePositionEvent` for the new rendering. This way, we would have something clearly distinct from some expected behavior.

@AxelRICHARD will bring support for the `DoubleClickTool` soon which has the advantage of doing exactly what it says